### PR TITLE
Add nullable option

### PR DESCRIPTION
**Pull request by VladimirPal**
https://github.com/flask-restful/flask-restful/issues/432

### CODE REVIEW

File: flask_restful/reqparse.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ class Argument(object):
         returned by :meth:`~reqparse.RequestParser.parse_args()`.
     :param bool required: Whether or not the argument may be omitted (optionals
         only).
+    :param bool nullable: Whether or not the argument may be None type (optionals
```
Comment (joshfriend) [inline flask_restful/reqparse.py]: I'm confused. You said this was an error with `required=True` but the docs you added say otherwise?

**Comment by joshfriend** [comment]
Just to see if I understand what the issue is here, given a request parser:

```python
parser = RequestParser()
parser.add_argument('foo', required=True)
```

And request data:

```json
{
    "foo": null
}
```

This raises `BadRequest` with an error about the `foo` argument being missing?

I would agree that this seems incorrect, though I don't really like how the correction is disabled by default. However, it would preserve the existing behavior, so it should probably stay that way.

Please add some tests to excercise this new behavior.

File: flask_restful/reqparse.py (hunk 1 above)
Comment (VladimirPal) [inline flask_restful/reqparse.py]: @joshfriend It said @huxuan in this issue https://github.com/flask-restful/flask-restful/issues/432 I don't think that it's an required error, required parameter works well. I'm only added new parameter - nullable.

**Comment by VladimirPal** [comment]
@joshfriend About default you are right, i will fix it and will add tests.

**Comment by huxuan** [comment]
Thanks for @VladimirPal ‘s attention about my issue, but there seems to have some misunderstanding. Actually the problem is `required=True` doesn't do what it should be. When we request with 
```
GET http://localhost:5000/api/test?foo=&bar=1
```
`parser.add_argument('foo', required=True)` won't do anything about this kind of behavior. After some investigation, I think `required=True` should also handle this kind of request. What do you think of it? @VladimirPal @joshfriend

**Comment by VladimirPal** [comment]
Really it seems we have some misunderstanding. I'm undersand you @huxuan but my problem was like that:

```
parser.add_argument('end_datetime', type=inputs.date, required=False)
// or 
parser.add_argument('end_datetime', type=inputs.date)
```

With request:
```
{
    "end_datetime": null
}
```
And it raises BadRequest.

In db i have field like this:
```
end_datetime = db.Column(db.DateTime, nullable=True)
```

So i need to have ability to get nullable value.

**Comment by VladimirPal** [comment]
@joshfriend It seems my problem and this pr not relevant anymore. I just now installed this package through github master branch and i haven't my issue anymore.

**Pull request closed without merging**